src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -579,13 +579,20 @@
   (with-open [r (reader pem)]
     (SSLUtils/pemToCRLs r)))
 
+(schema/defn ^:always-validate crl-issued-by-cert? :- schema/Bool
+  "Given a CRL and a certificate, determine whether the CRL was issued by the
+  certificate."
+  [crl :- X509CRL
+   cert :- X509Certificate]
+  (= (.getIssuerX500Principal crl)
+     (.getSubjectX500Principal cert)))
+
 (schema/defn ^:always-validate pem->ca-crl :- X509CRL
   "Given a CRL chain and CA certificate, extract the CRL issued by the
   certificate"
   [crl-chain :- Readerable
    ca-cert :- X509Certificate]
-  (let [match-ca-cert? (fn [crl] (= (.getIssuerX500Principal crl)
-                                    (.getSubjectX500Principal ca-cert)))
+  (let [match-ca-cert? (fn [crl] (crl-issued-by-cert? crl ca-cert))
         crls (pem->crls crl-chain)
         crl (first (filter match-ca-cert? crls))]
```

Comment:
Side note, this is some pretty good evidence that the order of the CRLs does not need to matter.

Reply:
yeah, I saw that, but wasn't sure if there were other places we were making assumptions. like, maybe this lib doesn't care, but we care in puppetserver? idk.

Reply:
Yeah. When you do the check, be sure to try revoking a cert, make sure that code path doesn't have any gotchas.